### PR TITLE
MAT-7185: refactor library drafting

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE PR

Jira Ticket: [MAT-7185](https://jira.cms.gov/browse/MAT-7185)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included and Code coverage is verified.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java [hunk 1]
```diff
@@ -165,8 +165,7 @@ public ResponseEntity<CqlLibrary> createDraft(
       @Validated(CqlLibrary.ValidationSequence.class) @RequestBody final CqlLibraryDraft cqlLibrary,
       Principal principal) {
     var output =
-        versionService.createDraft(
-            id, cqlLibrary.getCqlLibraryName(), cqlLibrary.getCql(), principal.getName());
+        versionService.createDraft(id, cqlLibrary.getCqlLibraryName(), principal.getName());
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java:181]: I think we do not need  `CqlLibraryDraft cqlLibrary` dto if we use @RequestParam

Comment (adongare) [reply src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java:181]: or maybe you can ignore it.

File: src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java [hunk 2]
```diff
@@ -130,13 +130,16 @@ public CqlLibrary createDraft(String id, String cqlLibraryName, String cql, Stri
...
-    clonedCqlLibrary.setCql(cql);
     clonedCqlLibrary.setDraft(true);
     var now = Instant.now();
     clonedCqlLibrary.setCreatedAt(now);
     clonedCqlLibrary.setCreatedBy(username);
     clonedCqlLibrary.setLastModifiedAt(now);
     clonedCqlLibrary.setLastModifiedBy(username);
+    if (!cqlLibraryName.equals(cqlLibrary.getCqlLibraryName())) {
+      clonedCqlLibrary.setCql(
+          cqlLibrary.getCql().replace(cqlLibrary.getCqlLibraryName(), cqlLibraryName));
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java]: Thi is a generic replacement. The majority of the time this should work. I was thinking if we should use a lil bit strict replacement.

Comment (chubert-sb) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java]: I updated it so the first line would always change similarly to how it was happening in the frontend

**Review by adongare — APPROVED** [review]

**Pull request merged**